peer.go

```diff
@@ -71,3 +95,35 @@ func (peer enetPeer) SendPacket(packet Packet, channel uint8) error {
 	)
 	return nil
 }
+
+func (peer enetPeer) SetData(data []byte) {
+	if len(data) > MaxPeerDataLength {
+		panic(fmt.Sprintf("cannot store data with len > %d", MaxPeerDataLength))
+	}
+
+	if data == nil {
+		peer.cPeer.data = nil
+		return
+	}
+
+	// First byte is how long our slice is.
+	b := make([]byte, len(data)+1)
+	b[0] = byte(len(data))
+	copy(b[1:], data)
+	peer.cPeer.data = unsafe.Pointer(&b[0])
+}
+
+func (peer enetPeer) GetData() []byte {
+	ptr := unsafe.Pointer(peer.cPeer.data)
+
+	if ptr == nil {
+		return nil
+	}
+
+	return C.GoBytes(
+		// Skip the first byte as this is the slice length.
+		unsafe.Add(ptr, 1),
```

Comment:
This (to me) implies we might not want to use slices for this, but I'm not too familiar with the `unsafe` package.

Reply:
Hmm good point. At the time I wanted something that I could easily convert between C and go, but actually as this is all in Go memory, I'm now wondering if this couldn't just be any data that you wanted to pass (i.e. `any`). My memory is a little hazy.

I'll go and try and come up with something better and/or find better justifications for what's here.

Thanks for taking the time for a review. Appreciated.

Reply:
I spent a bit of time playing around with the idea of these functions accepting/returning `any`, but couldn't find a neat way to do this without violating cgo's memory rules (i.e. don't store Go pointers in C heap). I like the slice here because of the `C.CBytes` and `C.GoBytes` helpers. It did cross my mind to try serialise `any` in to `bytes` ([gob?](https://pkg.go.dev/encoding/gob)) before putting in C, but I wasn't sure this is something I wanted to get in to for this API - feels like a bit of a rabbit hole.

My use case for these changes is to store an application identifier alongside the peer; essentially my calling code would be `ev.GetPeer().SetData([]byte(userId))`. I feel like the `[]byte` is simple & flexible.

Reply:
Indeed, I think `[]byte` is sufficient here. This lets the user decide by themselves how stuff is serialized (for example using json)